pkg/services/searchV2/index.go

```diff
@@ -124,7 +135,17 @@ func (i *dashboardIndex) buildOrgIndex(ctx context.Context, orgID int64) (int, e
 	orgSearchIndexLoadTime := time.Since(started)
 	i.logger.Info("Finish loading org dashboards", "elapsed", orgSearchIndexLoadTime, "orgId", orgID)
 
-	reader, writer, err := initIndex(dashboards, i.logger)
+	uids := make([]string, 0, len(dashboards))
+	for _, d := range dashboards {
+		uids = append(uids, d.uid)
+	}
+
+	documentExtenders := make([]ExtendDocumentFunc, len(i.extenders))
+	for _, extender := range i.extenders {
+		documentExtenders = append(documentExtenders, extender.GetDocumentExtender(orgID, uids))
```

Comment:
I am not wild about needing to initialize this with *all* uids in the index (10k!) it seem we have two modes:
1. we are about to reindex everything
2. we are going to index one thing

perhaps 'empty' means everything?  then you can do SELECT * rather than 10k requests :)

Reply:
the query is doing `IN (uids...)` so it's one query in both cases, but i agree that it's probably more efficient to do `select *` when re-indexing everything 🤔

Reply:
here's an example implementation: https://github.com/grafana/grafana-enterprise/blob/7dcaf2f1d399308a06849f6b1c704151e49efb3c/src/pkg/extensions/searchV2/dashboard/extender.go#L29-L41